### PR TITLE
Added Timeout to HttpClientBuilder as DelegatingHandler

### PR DESCRIPTION
**Pull request by OttoVT**

### CODE REVIEW

File: src/Lykke.HttpClientGenerator/HttpClientGeneratorBuilder.cs [hunk 1]
```diff
@@ -82,6 +83,16 @@ public HttpClientGeneratorBuilder WithoutCaching()
             return this;
         }
 
+        /// <summary>
+        /// Configures to use timeout.
+        /// </summary>
+        /// <exception cref="System.TimeoutException">Throws for any client;s generated method</exception>
```
Comment (polonzer) [inline src/Lykke.HttpClientGenerator/HttpClientGeneratorBuilder.cs]: >client;s
Typo :)

File: src/Lykke.HttpClientGenerator/Infrastructure/TimeoutHandler.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Lykke.HttpClientGenerator.Infrastructure
+{
+    /// <summary>
+    /// Adds User-Agent header to the request
```
Comment (polonzer) [inline src/Lykke.HttpClientGenerator/Infrastructure/TimeoutHandler.cs]: Copy-paste.

**Review by polonzer — DISMISSED** [review]

File: src/Lykke.HttpClientGenerator/HttpClientGenerator.cs [hunk 3]
```diff
@@ -35,6 +36,13 @@ public static HttpClientGeneratorBuilder BuildForUrl(string rootUrl)
             return new HttpClientGeneratorBuilder(rootUrl);
         }
 
+        /// <inheritdoc />
+        public HttpClientGenerator(string rootUrl, IEnumerable<ICallsWrapper> callsWrappers,
+            IEnumerable<DelegatingHandler> httpDelegatingHandlers, TimeSpan? timeout = null)
```
Comment (KonstantinRyazantsev) [inline src/Lykke.HttpClientGenerator/HttpClientGenerator.cs]: Make timeout not optional. At the moment this ctor is ambigous with ctor at line 47 when timeout is not specified

Comment (KonstantinRyazantsev) [reply src/Lykke.HttpClientGenerator/HttpClientGenerator.cs]: And why timeout parameter is not used?

Comment (KonstantinRyazantsev) [reply src/Lykke.HttpClientGenerator/HttpClientGenerator.cs]: I've found out that this ctor is not used at all, remove it.

File: src/Lykke.HttpClientGenerator/Infrastructure/TimeoutHandler.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                {
+                    var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(cts.Token, cancellationToken);
+
+                    return await base.SendAsync(
+                        request,
+                        linkedCts.Token);
+                }
+                catch (OperationCanceledException)
+                {
+                    throw new TimeoutException();
```
Comment (KonstantinRyazantsev) [inline src/Lykke.HttpClientGenerator/Infrastructure/TimeoutHandler.cs]: pass catched exception as inner

**Review by KonstantinRyazantsev — APPROVED** [review]

**Review by polonzer — APPROVED** [review]

**Pull request merged**